### ISSUE TITLE
Usage with Nx

### ISSUE THREAD

**Issue by danjor** [daniel-sc/ng-extract-i18n-merge#23]
Hello,

Could it be possible to support Nx by at least be able to handle both angular.json and workspace.json ?
I'm using a workspace.json file instead of angular.json and target extract-i18n-merge is not added when installing ng-extract-i18n-merge package, so I can't use that command.

yarn run [PROJECT_ID]:extract-i18n-merge
yarn nx run [PROJECT_ID]:extract-i18n-merge

produce  

NX   Cannot find target 'extract-i18n-merge' for project [PROJECT_ID]

Thanks

**Comment by danjor** [comment]
I manage to have the target added by running 

```shell
yarn add -W ng-extract-i18n-merge && yarn nx g ng-extract-i18n-merge:ng-add
```

**Comment by daniel-sc** [comment]
@danjor thanks for bringing this up!
I admit, that I did not work with nx yet .. to get this working seamlessly with nx, I'd need some concrete requirements. Could you help with this? (additionally I'd help if you could share your project layout (files/folders) and workspace.json)

**Comment by davidyoung77** [comment]
`"localize": "nx affected --target=extract-i18n-merge",`

This is what I use for NX and it works perfectly

**Comment by maxbeba** [comment]
I am using this with nx workspace. Works perfect and supports nx affected. 

```
"extract-i18n": {
      "executor": "ng-extract-i18n-merge:ng-extract-i18n-merge",
      "options": {
        "browserTarget": "web:build",
        "format": "xlf",
        "outputPath": "apps/web/src/locales",
        "targetFiles": ["messages.de-DE.xlf"]
      }
    },
```

**Comment by LeLunZ** [comment]
Yes I am also using a similar config and everything works. (You need these in project.json)

And then you can just do:
nx run <project>:extract-i18n

**Comment by LeLunZ** [comment]
@mmart1n weird that it didn't work with `@nx/angular:application`, as it should be a direct drop in replacement for "@angular-devkit/build-angular:application".

**Comment by LeLunZ** [comment]
> [@danjor](https://github.com/danjor) thanks for bringing this up! I admit, that I did not work with nx yet .. to get this working seamlessly with nx, I'd need some concrete requirements. Could you help with this? (additionally I'd help if you could share your project layout (files/folders) and workspace.json)

@daniel-sc 
In nx the `ng` command shouldn't be used, so when I first read this I wasn't actually sure what the problem was.
Normally if you use nx, and an automatic setup should happen with "ng add" you need to manually do that setup if no "generator" is provided.

To also do the automatic setup in nx, a `generator` (similar to angular schematics) is required.
This generator can be published as part of a nx plugin. 

If there would be an plugin (and a generator) for this package someone could just run "nx add ng-exctract-i18n-merge" and everything should be setup as expected. Similar to the automatic setup in a normal angular based project.

[Here](https://github.com/nrwl/nx/blob/master/community/approved-plugins.json) is a list of all the community plugins, where someone could check how others have setup these nx plugins. And here is the [nx documentation](https://nx.dev/extending-nx/recipes/local-generators).

---

To simplify the setup for someone using nx,  I think its also okay, to add a minimal example setup configuration, and link to it from the documentation or show how the run target should look like in the documentation .

**Comment by mmart1n** [comment]
> [@mmart1n](https://github.com/mmart1n) weird that it didn't work with `@nx/angular:application`, as it should be a direct drop in replacement for "@angular-devkit/build-angular:application".

It doesn't throw any errors, but it generates empty XLF files. For now, I'm using the workaround with the custom builder configuration for i18n, but I believe there should be a more straightforward solution.

**Comment by daniel-sc** [comment]
Hello everyone, great to see you are helping out each other!

I personally won't create an nx generator or specific documentation, but I'd be happy to include links to a matching nx generator and/or documentation/blog posts in the readme (or even a short section in the readme itsself regarding nx setup). I'd try to support as much as possible, if someone steps up and drives this topic!
